pkg/sql/scrub_constraint.go

```diff
@@ -96,8 +97,26 @@ func (o *sqlCheckConstraintCheckOperation) Start(params runParams) error {
 		}
 	}
 
-	rows, err := params.extendedEvalCtx.ExecCfg.InternalExecutor.Query(
-		ctx, "check-constraint", params.p.txn, tree.AsStringWithFlags(sel, tree.FmtParsable),
+	override := sqlbase.InternalExecutorSessionDataOverride{
+		User: security.RootUser,
+	}
+	isTempSchema, clusterWideID, err := temporarySchemaSessionID(tn.Schema())
+	if err != nil {
+		return err
+	}
+	if isTempSchema {
```

Comment:
shouldn't we just always put the temp schema in the search path here? Or rather, why isn't the temp schema in the search path for this query?

Reply:
hmm, so for some reason the session data is not set for the internal executor here.

Reply:
i'm going to close this as i think the fix is deeper rooted.